File: kotlin/src/test/kotlin/com/loomup/client/RealtimeUrlTest.kt
package com.loomup.client

import kotlin.test.Test
import kotlin.test.assertEquals

class RealtimeUrlTest {
    @Test
    fun realtimeWebSocketUrlPreservesBasePath() {
        val cases = listOf(
            "https://tryloomup.com/p/project-id" to "wss://tryloomup.com/p/project-id/realtime",
            "http://localhost:3000" to "ws://localhost:3000/realtime",
            "https://tryloomup.com/p/project-id/" to "wss://tryloomup.com/p/project-id/realtime",
        )

        cases.forEach { (base, expected) ->
            assertEquals(expected, realtimeWebSocketUrl(base))
        }
    }
}

File: flutter/test/realtime_url_test.dart
import 'package:loomup/loomup.dart';
import 'package:test/test.dart';

void main() {
  test('realtime WebSocket URL preserves the base path', () {
    final cases = {
      'https://tryloomup.com/p/project-id':
          'wss://tryloomup.com/p/project-id/realtime',
      'http://localhost:3000': 'ws://localhost:3000/realtime',
      'https://tryloomup.com/p/project-id/':
          'wss://tryloomup.com/p/project-id/realtime',
    };

    for (final entry in cases.entries) {
      expect(
        realtimeWebSocketUrl(Uri.parse(entry.key)).toString(),
        entry.value,
      );
    }
  });
}
